Content.Server/Anomaly/AnomalySystem.Behaviour.cs

```diff
@@ -0,0 +1,104 @@
+using System.Linq;
+using Content.Shared.Anomaly.Components;
+using Content.Shared.Anomaly.Prototypes;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.Manager;
+
+namespace Content.Server.Anomaly;
+public sealed partial class AnomalySystem
+{
+    [Dependency] private readonly IComponentFactory _componentFactory = default!;
+    [Dependency] private readonly ISerializationManager _serialization = default!;
+
+    private List<AnomalyBehaviourPrototype> _behaviourList = new();
+    private void InitializeBehaviour()
+    {
+        SubscribeLocalEvent<PrototypesReloadedEventArgs>(OnPrototypesReloaded);
+
+        //Cache all behaviors into a list at the beginning of the round
+        _behaviourList.AddRange(_prototype.EnumeratePrototypes<AnomalyBehaviourPrototype>());
+    }
+
+    private void OnPrototypesReloaded(PrototypesReloadedEventArgs ev)
+    {
+        _behaviourList.Clear();
+        _behaviourList.AddRange(_prototype.EnumeratePrototypes<AnomalyBehaviourPrototype>());
+    }
+
+    private string GetRandomBehaviour()
+    {
+        var totalWeight = _behaviourList.Sum(x => x.Weight);
+        var randomValue = _random.NextFloat(totalWeight);
+
+        foreach (var b in _behaviourList)
+        {
+            if (randomValue < b.Weight)
+            {
+                return b.ID;
+            }
+
+            randomValue -= b.Weight;
+        }
+        // Shouldn't happen
+        throw new InvalidOperationException($"Invalid weighted variantize anomaly behaviour pick!");
+    }
```

Comment:
Please don't copy-paste code out of random, I recognise that comment because I wrote it.